### DISCUSSION TITLE
Order of commits in git log v. git rebase -i

### DISCUSSION THREAD

**Discussion by NadChel** [community/community#69869] [Programming Help] [answered]
### Body

When I execute `git log --oneline`, I get
```
3c500bd (HEAD -> fix/AFI-4693-fix_bad_request_response, origin/fix/AFI-4693-fix_bad_request_response) Merge remote-tracking branch 'origin/fix
/AFI-4693-fix_bad_request_response' into fix/AFI-4693-fix_bad_request_response
3bb9f25 credit AFI-4693 fix bad request response + fix CardMapper, CardMapperTest
e0c11d9 credit AFI-4693 fix bad request response + fix CardMapper, CardMapperTest
5cd3dbe credit AFI-4693 fix bad request response + fix CardMapper, CardMapperTest
813f105 (dev) Merge remote-tracking branch 'origin/release' into dev
91b2e4b (origin/release) Merge branch 'release_migration-fix' into 'release'
80ffd72 (origin/release_migration-fix) {credit}{hotfix}{change V202302210000__update.sql and fix migration checksum}
93c1df9 Merge branch 'release_migration-fix' into 'release'
071f998 {credit}{hotfix}{change V202302210000__update.sql and fix migration checksum}
58a8c72 Merge branch 'release_migration-fix' into 'release'
ce69851 {credit}{hotfix}{change V202302210000__update.sql and fix migration checksum}
99fbdf6 Merge branch 'fix_migration' into 'release'
f003d9b (origin/dev, origin/HEAD) Merge branch 'revert-3525a1a3' into 'dev'
deeba0e Revert "Merge branch 'fix_migration' into 'dev'"
```
When I execute `git rebase -i HEAD~2`, I get this. The order and the number of commits seems completely random relative to the log above. Also, I don't see why `HEAD~2` printed out five commits. How exactly does it work?
```
pick 3bb9f25 credit AFI-4693 fix bad request response + fix CardMapper, CardMapperTest
pick ce69851 {credit}{hotfix}{change V202302210000__update.sql and fix migration checksum}
pick 071f998 {credit}{hotfix}{change V202302210000__update.sql and fix migration checksum}
pick 80ffd72 {credit}{hotfix}{change V202302210000__update.sql and fix migration checksum}
pick 5cd3dbe credit AFI-4693 fix bad request response + fix CardMapper, CardMapperTest
```

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by fredschuck** [comment]
I think that you are only seeing the two commits and their ancestors in HEAD~2. Hope that helps :)

**Comment by RomanchenkoAS** [comment, marked as answer]
The behavior you're seeing may be because git rebase -i HEAD~2 shows not just your own commits but also other commits that are reachable from HEAD. You may see other commits that are part of branches you've merged into your own. This list of commits is what would be moved if you proceed with the rebase operation. Rebasing effectively replays commits on top of a point you choose, so the list of commits you're seeing is a sequence that will be replayed.

If you want to rebase only your own commits, you need to identify the range

git rebase -i <commit_hash>^..HEAD
